l10n_cl_invoicing_policy/models/res_partner.py

```diff
@@ -23,21 +26,24 @@ def _default_invoicing_policy(self):
         default=_default_invoicing_policy,
         help="""
         * Ticket: Only for individuals. 1 invoice for each delivery order.
-        * Invoice: Only for companies. The VAT is required. 1 invoice for each
-          delivery order. Requires the customer PO # on the SO/Invoice.
-        * Electronic Guide: Only for companies. 1 invoice at the end of the
-          month for all the delivery orders of that month.""")
+        * Invoice: Only for companies and their children. The VAT is required.
+          1 invoice for each delivery order.
+          Requires the customer PO # on the SO/Invoice.
+        * Electronic Guide: Only for companies and their children.
+          1 invoice at the end of the month for all the delivery orders of
+          that month.""")
 
-    @api.constrains('is_company', 'invoicing_policy')
+    @api.constrains('is_company', 'invoicing_policy', 'parent_id')
     def check_invoicing_policy(self):
-        if self.invoicing_policy == 'ticket':
-            if self.is_company:
-                raise UserError(_('The invoicing policy Ticket only applies '
-                                  'to individuals.'))
-        else:
-            if not self.is_company:
-                raise UserError(_('The selected invoicing policy only applies'
-                                  ' to companies.'))
+        rec = self
+        while rec.parent_id:
+            rec = rec.parent_id
+        if rec.is_company and self.invoicing_policy == 'ticket':
+            raise UserError(_(
+                'The invoicing policy Ticket only applies to individuals.'))
+        if not rec.is_company and self.invoicing_policy != 'ticket':
+            raise UserError(_(
+                'The selected invoicing policy only applies to companies.'))
```

Comment:
You are rechecking the parent invoicing policy every time when adding/editing a child address.
Seems redundant to me.
Why not just check on parent edits?